src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -292,36 +306,93 @@
 (def required-cli-options
   [:config])
 
+(defn validate-options
+  [options action-on-error-fn]
+  (cond
+    (and (contains? options :runinterval)
+         (contains? options :nummsgs))
+    (do
+      (utils/println-err
+        "Error: -N/--nummsgs runs immediately and is not compatable with -i/--runinterval")
+      (action-on-error-fn))
+
+    (and (contains? options :archive)
+         (some #{:reports :catalogs :facts :default} (keys options)))
+    (do
+      (utils/println-err
+        "Error: -A/--archive is incompatible with -F/--facts, -C/--catalogs, -R/--reports, -D/--default.")
+      (action-on-error-fn))
+
+    (and (contains? options :default)
+         (some #{:reports :catalogs :facts :archive} (keys options)))
+    (do
+      (utils/println-err
+        "Error: -D/--archive is incompatible with -F/--facts, -C/--catalogs, -R/--reports, -A/--archive.")
+      (action-on-error-fn))
+    :else options))
+
+(defn default-options
+  [options]
+  (-> options
+      (assoc :facts "puppetlabs/puppetdb/benchmark/samples/facts")
+      (assoc :reports "puppetlabs/puppetdb/benchmark/samples/reports")
+      (assoc :catalogs "puppetlabs/puppetdb/benchmark/samples/catalogs")
+      (assoc :from-cp? true)))
+
 (defn- validate-cli!
   [args]
   (try+
-   (ks/cli! args supported-cli-options required-cli-options)
-   (catch map? m
-     (println (:message m))
-     (case (:type m)
-       :puppetlabs.kitchensink.core/cli-error (System/exit 1)
-       :puppetlabs.kitchensink.core/cli-help (System/exit 0)))))
-
-(defn validate-nummsgs [options action-on-error-fn]
-  (when (and (contains? options :runinterval)
-             (contains? options :nummsgs))
-    (utils/println-err "Error: -N/--nummsgs runs immediately and is not compatable with -i/--runinterval")
-    (action-on-error-fn)))
+    (let [options (-> (ks/cli! args supported-cli-options required-cli-options)
+                      first
+                      (validate-options #(System/exit 1)))]
+      (if (:default options)
+        (default-options options)
+        options))
+
+    (catch map? m
+      (println (:message m))
+      (case (:type m)
+        :puppetlabs.kitchensink.core/cli-error (System/exit 1)
+        :puppetlabs.kitchensink.core/cli-help (System/exit 0)))))
+
+(defn process-tar-entry
+  [archive-path tar-reader]
+  (fn [acc entry]
+    (let [path (.getName entry)
+          catalog-pattern "catalogs.*\\.json$"
```

Comment:
Seems like you could let bind this outside of the fn and push the re-pattern creation there too, then you wouldn't need to recreate these regex patterns on each file in the tarball.